src/rate/index.tsx

```diff
@@ -0,0 +1,236 @@
+import { DNode, RenderResult } from '@dojo/framework/core/interfaces';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import focus from '@dojo/framework/core/middleware/focus';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import Icon from '../icon';
+import Radio from '../radio';
+import RadioGroup, { RadioGroupChildren } from '../radio-group';
+import theme from '../middleware/theme';
+import * as baseCss from '../common/styles/base.m.css';
+import * as iconCss from '../theme/default/icon.m.css';
+import * as radioCss from '../theme/default/radio.m.css';
+import * as radioGroupCss from '../theme/default/radio-group.m.css';
+import * as css from '../theme/default/rate.m.css';
+import bundle from './nls/Rate';
+
+export interface RateProperties {
+	/** The name attribute for this rating group */
+	name: string;
+	/** The label to be displayed in the legend */
+	label?: string;
```

Comment:
As @tomdye already pointed out, making `label` a child renderer would be more consistent with what we've been doing for the other widgets:

```
<Rate>
  {{
    label: 'Rating',
    character: (filled) => (
      <span>{filled ? <Icon type="upIcon" /> : <Icon type="downIcon" />}</span>
    )
  }}
</Rate>
```

Reply:
Yep. As soon as `Radio` is updated.